livre/core/data/MemoryUnit.h

```diff
@@ -120,6 +120,12 @@ class AllocMemoryUnit : public MemoryUnit, public boost::noncopyable
         ::memcpy( _rawData.getData(), sourceData, size * sizeof( T ) );
     }
 
+    void allocAndSetData( const void* sourceData, const size_t size )
+    {
+        alloc( sizeof( uint8_t ) * size );
+        ::memcpy( _rawData.getData(), sourceData, size * sizeof( uint8_t ) );
+    }
```

Comment:
use partial template specialization of previous method (forward to allocAndSetData< uint8_t >())